### PR TITLE
Resolve symbol in TypeChecker#getConstValue

### LINKED ISSUE #4217: TypeChecker#getConstantValue(node) doesn't properly resolve values
Courtesy of @ivogabe from [here](https://gitter.im/Microsoft/TypeScript?at=55c4bdfdcdd8bb455f2f8c7e).

> I'm working with the compiler api and I noticed that `TypeChecker#getConstantValue(node)` only returns a value if I first call `TypeChecker#getSymbolAtLocation(node)`. `node` is an ElementAccessExpression or a PropertyAccessExpression. Is that a bug or by design?

### PR DESCRIPTION
**Pull request by ivogabe**
Fixes #4217

### CODE REVIEW

**Comment by ivogabe** [comment]
This seems to break normal compilation, will try to figure it out.

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -11707,7 +11707,7 @@ module ts {
                 return getEnumMemberValue(<EnumMember>node);
             }
 
-            let symbol = getNodeLinks(node).resolvedSymbol;
+            let symbol = getSymbolAtLocation(node);
```
Comment (vladima) [inline src/compiler/checker.ts]: `getSymbolAtLocation` is a top language service API that should not be used on the checker itself. For this particular case I think you should use `getResolvedSymbol`. Also implementation of `getResolvedSymbol` should be augmented (i.e. by adding an extra argument) so in this case it won't report an error when name cannot be resolved

Comment (vladima) [inline src/compiler/checker.ts]: ok, I guess I'm wrong since node here will be `PropertyAccessExpression` / `ElementAccessExpression` not `Identifier`

Comment (ivogabe) [inline src/compiler/checker.ts]: Would it be easier to split `getConstantValue` into two functions: an internal and an external one? The external function can then call `getSymbolAtLocation`

**Comment by ivogabe** [comment]
Closing this as this breaks compilation and still doesn't fix the issue for bracket notation.

**Pull request closed without merging**